app/controllers/concerns/sortable.rb

```diff
@@ -0,0 +1,24 @@
+module Sortable
+  extend ActiveSupport::Concern
+
+  def sort
+    ActiveRecord::Base.transaction do
+      sort_params[:sortable_ids].each_with_index do |id, index|
+        sortable_records.where(id: id.to_i).update_all({ position: index + 1, updated_at: Time.now })
```

Comment:
The `to_i` is needed because it's coming as a list of strings from the front end

We need to check agains the list of permitted records from the controller first, so I updated this to check the list of permitted ids and update if the id is included
```ruby
if sortable_records[:ids].include?(id)
  klass.update(id, position: index)
end
```